src/google-classroom-extractor/edfi_google_classroom_extractor/request.py

```diff
@@ -14,15 +14,15 @@
 from edfi_google_classroom_extractor.api.teachers import request_all_teachers_as_df
 from edfi_google_classroom_extractor.api.submissions import request_all_submissions_as_df
 from edfi_google_classroom_extractor.api.usage import request_all_usage_as_df
+from edfi_google_classroom_extractor.helpers.arg_parser import MainArguments
 from .result import Result
 
 
 def request_all(
     classroom_resource: Resource,
     reports_resource: Resource,
     sync_db: sqlalchemy.engine.base.Engine,
-    env_usage_start_date: str,
-    env_usage_end_date: str,
+    arguments: MainArguments
```

Comment:
Now that I look at this more closely, I am very surprised to see use of `MainArguments` here. Normally that should only be used by the facade class, not here in this request class. The if/then statements below are also very strange to me - the purpose of the facade is to handle that logic. The request class should not need to handle that logic.